### PR TITLE
 Path: Add ordering comparison functions 

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Path: Add ordering comparison functions

The ordering of path (as obtained when iterating over a directory) in Littlefs
is not exactly what is expected.

This implementation contains 2 comparision functions, one matching what is expected,
and one matching the iteration order of littlefs directories,
as described in https://github.com/littlefs-project/littlefs/issues/923

The fact that directories are ordered is documented:
https://github.com/littlefs-project/littlefs/blob/f53a0cc961a8acac85f868b431d2f3e58e447ba3/SPEC.md?plain=1#L304

### CODE REVIEW

File: src/path.rs [hunk 1]
```diff
@@ -750,4 +815,60 @@ mod tests {
...
+        );
+        assert_eq!(
+            path!("some_path_b").cmp_lfs(path!("some_path")),
+            Ordering::Less
+        );
+        assert_eq!(
+            path!("some_path").cmp_lfs(path!("some_path")),
+            Ordering::Equal
+        );
+    }
```
Comment (robin-nitrokey) [inline src/path.rs:873]: Isn’t this the same as the tests in the doc comments?

File: src/path.rs [hunk 2]
```diff
@@ -20,6 +23,66 @@ pub struct Path {
...
+            Ordering::Greater => Ordering::Greater,
+            Ordering::Equal if this.len() != other.len() => {
+                if this.len() < other.len() {
+                    Ordering::Greater
+                } else {
+                    Ordering::Less
+                }
+            }
+            Ordering::Equal => Ordering::Equal,
+        }
```
Comment (robin-nitrokey) [inline src/path.rs:82]: could also be written more concise as:

```rust
this[..min_len]
  .cmp(&other[..min_len])
  .then_with(|| other.len().cmp(&this.len()))
```

not sure if this is really an improvement though

**Review by robin-nitrokey — APPROVED** [review]

File: src/path.rs (hunk 1 above)
Comment (sosthene-nitrokey) [reply src/path.rs:873]: Yes. You're right, this means it can be removed from "normal" tests.

File: src/path.rs (hunk 2 above)
Comment (robin-nitrokey) [reply src/path.rs:82]: I agree that it unfortunately is not an intuitive name, which is especially bad as I think the control flow is much clearer than in the match case.

Comment (robin-nitrokey) [reply src/path.rs:82]: another suggestion:

```rust
        match this[..min_len].cmp(&other[..min_len]) {
            Ordering::Less => Ordering::Less,
            Ordering::Greater => Ordering::Greater,
            Ordering::Equal => other.len().cmp(&this.len()),
        }
```

Comment (sosthene-nitrokey) [reply src/path.rs:82]: Yeah this would actually match the error made in the C implementation.

**Pull request merged**